src/Compilers/Core/MSBuildTask/BuildClient.cs

```diff
@@ -31,11 +31,17 @@ internal static class BuildClient
         private const int TimeOutMsNewProcess = 20000; 
 
         /// <summary>
-        /// Try to get the directory this assembly is in.
+        /// Try to get the directory this assembly is in. Returns null if assembly
+        /// was in the GAC.
         /// </summary>
-        private static string GetClientDir()
+        private static string TryGetClientDir()
         {
-            var uri = new Uri(Assembly.GetExecutingAssembly().CodeBase);
+            var assembly = Assembly.GetExecutingAssembly();
```

Comment:
Having said that, perhaps we should really be using Assembly.GetEntryAssembly()
This will give us the path to MSBuild.exe, csc.exe, etc.
It would be odd (although possible, I suppose) to put EXEs in the GAC.